libgloss/patmos/open.c

```diff
@@ -29,15 +29,35 @@
 //   policies, either expressed or implied, of the copyright holder.
     
 #include <errno.h>
+#include <stdarg.h>
 
 #undef errno
 extern int  errno;
 
+//******************************************************************************
+/// patmos-plug: open: stub called if 'patmosplug_open(const char *, int, int)'
+/// is not defined.
+int _patmosplug_open(const char *name, int flags, int mode) {
+  errno = ENOSYS;
+  return -1;
+}
+
+int patmosplug_open(const char *name, int flags, int mode)
+    __attribute__((weak, alias("_patmosplug_open")));
+
 //******************************************************************************
 /// _open - open a file.
-int _open(const char *name, int flags, int mode)
+int _open(const char *name, int flags, ...)
```

Comment:
Could maybe use a comment on why it is necessary to change the signature of this function to include a vararg. Just to make sure that no-one changes it back at some point, not realizing the original intent.